### DISCUSSION TITLE
Is there a way to toggle quirks via MQTT or API?

### DISCUSSION THREAD

**Discussion by rohankapoorcom** [Hypfer/Valetudo#1832] [Q&A / Support] [answered]
I have a Dreame Z10 pro that struggles in one of my rooms with a darker carpet. I’ve found that turning off the “collision avoidance” quirk before cleaning that room generally results in better performance. I don’t want to leave it with that quirk disabled all of the time otherwise it bumps into furniture in other rooms.

Is there a way that I could programmatically enable/disable the quirk through API or MQTT so I could have Home Assistant take care of it when it’s dispatching the vacuum to that room?

Thanks!

**Comment by Hypfer** [comment, marked as answer]
Please refer to the inbuilt help of the quirk feature. There's a question mark icon on the top right of that view that opens a modal that answers this exact question.

**Reply by rohankapoorcom** [reply]
Thanks - I had read through all of the documentation on the website and hadn't found it there, but didn't think to look in the UI itself. Appreciate the quick response.
